custom_components/yahoofinance/coordinator.py

```diff
@@ -257,67 +277,82 @@ def add_symbol(self, symbol: str) -> bool:
     async def get_json(self) -> dict:
         """Get the JSON data."""
 
-        url = BASE + ",".join(self._symbols)
-        cookies = None
-
-        crumb = self._cc.crumb
-        if crumb is None:
-            crumb = await self._cc.try_get_crumb()
-        if crumb is not None:
-            url = url + "&crumb=" + urllib.parse.quote(crumb)
-            cookies = self._cc.cookies
-
+        url = await self.build_request_url()
+        cookies = self._cc.cookies
         _LOGGER.debug("Requesting data from '%s'", url)
 
         try:
             async with async_timeout.timeout(WEBSESSION_TIMEOUT):
-                headers = {"User-Agent": USER_AGENT}
                 response = await self.websession.get(
-                    url, headers=headers, cookies=cookies
+                    url, headers=REQUEST_HEADERS, cookies=cookies

```

Comment:
If crumb contains a slash, will it be automatically encoded? I had urllib.parse.quote for that which you have removed.